app/src/main/java/org/simple/clinic/editpatient/deletepatient/DeletePatientScreen.kt

```diff
@@ -0,0 +1,124 @@
+import org.simple.clinic.widgets.DividerItemDecorator
+import org.simple.clinic.widgets.dp
+import javax.inject.Inject
+
+class DeletePatientScreen(context: Context, attrs: AttributeSet) : ConstraintLayout(context, attrs), UiActions, DeletePatientUi {
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  @Inject
+  lateinit var effectHandlerFactory: DeletePatientEffectHandler.Factory
+
+  private val viewRenderer = DeletePatientViewRenderer(this)
+
+  private val screenKey by unsafeLazy {
+    screenRouter.key<DeletePatientScreenKey>(this)
+  }
+
+  private val events = PublishSubject.create<DeletePatientEvent>()
+
+  private val delegate by unsafeLazy {
+    MobiusDelegate.forView(
+        events = events,
+        defaultModel = DeletePatientModel.default(screenKey.patientUuid),
+        init = DeletePatientInit(),
+        update = DeletePatientUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build(),
+        modelUpdateListener = viewRenderer::render
+    )
+  }
+
+  @SuppressLint("CheckResult")
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) return
+
+    context.injector<DeletePatientScreenInjector>().inject(this)
+
+    toolbar.setNavigationOnClickListener { screenRouter.pop() }
+    with(deleteReasonsRecyclerView) {
+      addItemDecoration(DividerItemDecorator(context, marginStart = 56.dp, marginEnd = 16.dp))
+    }
+  }
+
+  override fun onAttachedToWindow() {
+    super.onAttachedToWindow()
+    delegate.start()
+  }
+
+  override fun onDetachedFromWindow() {
+    delegate.stop()
+    super.onDetachedFromWindow()
+  }
+
+  override fun onSaveInstanceState(): Parcelable? {
+    return delegate.onSaveInstanceState(super.onSaveInstanceState())
+  }
+
+  override fun onRestoreInstanceState(state: Parcelable?) {
+    val viewState = delegate.onRestoreInstanceState(state)
+    super.onRestoreInstanceState(viewState)
+  }
+
+  override fun showDeleteReasons(patientDeleteReasons: List<PatientDeleteReason>, selectedReason: PatientDeleteReason?) {
+  }
+
+  override fun showConfirmDeleteDialog(patientName: String, deletedReason: DeletedReason) {
+    val title = when (deletedReason) {
+      DeletedReason.Duplicate -> context.getString(R.string.deletereason_duplicate_patient_confirm_title)
+      DeletedReason.AccidentalRegistration -> context.getString(R.string.deletereason_accidental_registration_confirm_title)
+      is DeletedReason.Unknown -> deletedReason.actualValue
+    }
+
+    val message = when (deletedReason) {
+      DeletedReason.Duplicate -> context.getString(R.string.deletereason_duplicate_patient_confirm_message, patientName)
+      DeletedReason.AccidentalRegistration -> context.getString(R.string.deletereason_accidental_registration_confirm_message, patientName)
+      else -> throw IllegalArgumentException("Unknown deleted reason: $deletedReason")
+    }
+
+    AlertDialog.Builder(context, R.style.Clinic_V2_DialogStyle_Destructive)
```

Comment:
We have been using dialog fragments so that they are retained upon screen restoration. Can you switch to using them like we do elsewhere?

Reply:
Seems like if I switch to dialog fragments, I have to either setup listeners for dialog actions using interfaces or Kotlin higher-order function or use its own Mobius loop _(I looked at other dialog fragment implementations in Simple and they are using a different a Mobius loop than the screen that opened the dialog)_. Should I use a different Mobius loop that handles the dialog actions? I am thinking of raising a different PR for it, WDYT?

Reply:
On second thought, I am thinking it's fine to leave it as is.

We have created Mobius loops for dialogs that are handling some business logic themselves. However in this case, it seems like the dialog just forwards events to the parent screen and logic is handled there.

Making this a dialog fragment would mean that we have to move the logic of deleting/marking patients as dead to the dialog class and creating a new mobius loop there since there is no easy way to reconnect events on state restoration and we'll have to check if the fragment exists or not before reattaching events.

So let's leave it as is. We might actually want to consider moving away from dialog fragments, looks like.